src/corelibs/U2View/src/ov_msa/MSAEditorSequenceArea.cpp

```diff
@@ -850,10 +852,19 @@ void MSAEditorSequenceArea::sl_enableFreeRowOrderMode(const QList<QStringList> &
         collapsibleGroupList << MaCollapsibleGroup(maRowIndexList, maRowIdList, isCollapsed);
     }
     editor->setRowOrderMode(MaEditorRowOrderMode::Free);
+    freeModeMasterMarkersSet.insert(marker);
     updateRowOrderActionsState();
     ui->getCollapseModel()->update(collapsibleGroupList);
 }
 
+void MSAEditorSequenceArea::disableFreeRowOrderMode(QObject *marker) {
+    freeModeMasterMarkersSet.remove(marker);
+    if (freeModeMasterMarkersSet.isEmpty() && editor->getRowOrderMode() == MaEditorRowOrderMode::Free) {
+        // Switch back to the Original ordering.
+        sl_toggleSequenceRowOrder(false);
```

Comment:
Original ordering is the same as in the file. But what if sequences were reordered during the `MaEditorRowOrderMode::Free` mode? The file will be updated? Or how it works like?

Reply:
'Free' is a virtual ordering managed by Tree viewer.
When MSA Editor is in Free mode the file is not updated (it is virtual like Order-By-Sequence).